internal/pkg/manifest/pipeline_stage.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store/ssm"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/workspace"
+)
+
+// PipelineStage represents configuration for each deployment stage
+// of a workspace. A stage consists of the Archer Environment the pipeline
+// is deloying to and the containerized applications that will be deployed.
+type PipelineStage struct {
+	*associatedEnvironment `yaml:",inline"`
+	LocalApplications      []string `yaml:"-"`
+}
+
+// associatedEnvironment defines the necessary information a pipline stage
+// needs for an Archer Environment.
+type associatedEnvironment struct {
+	// Name of the Archer project this environment belongs to
+	ProjectName string `yaml:"-"`
+
+	// Name of the environment, must be unique within a project.
+	// This is also the name of the pipeline stage.
+	Name string `yaml:"name"`
+	// T region this environment is stored in.
+	Region string `yaml:"-"`
+	// AccountID of the account this environment is stored in.
+	AccountID string `yaml:"-"`
+	// Whether or not this environment is a production environment.
+	Prod bool `yaml:"-"`
+}
+
+// associatedApp defines the name of an Archer application that will be
+// deploye to certain pipeline stage.
+type associatedApp struct {
+	Name string
+}
+
+var fetchAssociatedEnvAndApps func(stageName string) (*associatedEnvironment, []string, error) = func(stageName string) (*associatedEnvironment, []string, error) {
```

Comment:
Does this function need to be assigned to a variable?

Reply:
Previously it was done that way so that we can overwrite it in unit tests to avoid actually making calls to SSM.